tango_simlib/tango_launcher.py

```diff
@@ -63,11 +63,10 @@ def put_device_property(dev_name, property_name, property_value, file_name):
 def start_device(opts):
     server_name = os.path.basename(opts.server_command)
     number_of_devices = len(opts.name)
-    # Register tango devices if TANGO DB is being used
-    if not opts.file_name:
-        for i in range(number_of_devices):
-            register_device(
-                opts.name[i], opts.device_class[i], server_name, opts.server_instance)
+    # Register tango devices
+    for i in range(number_of_devices):
+        register_device(
```

Comment:
I'm not sure we're solving the correct problem here.  The `register_device` method instantiates a `Database()` object.  https://github.com/ska-sa/tango-simlib/blob/9308cf8/tango_simlib/tango_launcher.py#L51

Called with no parameters it will try to connect to a real TANGO database, using the TANGO_HOST environment variable for the connection details.  Try killing your database server `/usr/lib/tango/DataBaseds` and see if this script still works.

Reply:
Yes, that's right. The simulator failed to start when I killed the process. This is what I found:

When using file as a database, there's no need to register the devices in the database (they're already defined in the file). In fact, registering devices to file databases fails.
<img width="561" alt="screen shot 2018-07-27 at 06 29 49" src="https://user-images.githubusercontent.com/37175409/43301597-86ce13b0-9166-11e8-98de-3140f478c60d.png">

<img width="850" alt="screen shot 2018-07-27 at 06 31 36" src="https://user-images.githubusercontent.com/37175409/43301654-ce2f2942-9166-11e8-97bb-be5b8db253df.png">

In `tango_sim_generator`, the device names are retrieved from a real tango Database to run the simcontrol interface.
https://github.com/ska-sa/tango-simlib/blob/b72e486a1eff97bdb9dae6f469600c355f30e9d1/tango_simlib/tango_sim_generator.py#L458-L460
These devices get registered in the database only when tango_launcher runs. As such, we will have to register the device whether or not we use a file as the database.